### PR TITLE
remove sanitize-html and replace with dompurify

### PR DESCRIPTION
**Pull request by ostowe**
## Issue(s): Relates to or closes...
[**LEDE-1456**](https://alleyinteractive.atlassian.net/browse/LEDE-1456)

## Summary: This pull request will...
* Remove `sanitize-html` which has caused us continuous issues, replace with `dompurify` which is smaller and more in line with the current Alley standard.
* Fix storybook (I realize this has nothing to do with the intent of this PR, but some package updates necessitated it for it to continue working)
* Replace our DOM shim with `jsdom` as we need it anyway for `dompurify`
* Add a `sanitizeHTML` field to `irving.config.js` to allow users to add/modify sanitization configuration

## Tests: I know this code works because...
Tested locally and on lede

### CODE REVIEW

File: packages/core/utils/sanitizeHTML.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  {
+    ADD_TAGS: [
+      'iframe', 'link',
+    ],
+    ADD_ATTR: [
+      'rel', 'href', 'type', 'target', 'nohref', 'xml:lang', 'currenttime',
+      'duration', 'char', 'charoff', 'hspace', 'longdesc', 'vspace', 'autofocus',
+      'dirname', 'form', 'formaction', 'formenctype', 'formmethod',
+      'formnovalidate', 'formtaget', 'rules', 'abbr', 'axis',
+    ],
```
Comment (kaitlinbolling) [inline packages/core/utils/sanitizeHTML.js:34]: I think the following are included/not needed here: `rel`, `href`, `type`

A couple I think need to be added: `aria-*`, `data-*`, `buffered`, `intrinsicsize`

Also, might be nice to alphabetize this list for readability.

File: packages/core/utils/sanitizeHTML.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  sanitizeHTML = require('dompurify');
+}
+
+module.exports = sanitizeHTML;
+module.exports.config = getValueFromConfig(
+  'sanitizeHTML',
+  {
+    ADD_TAGS: [
+      'iframe', 'link',
+    ],
```
Comment (kaitlinbolling) [inline packages/core/utils/sanitizeHTML.js:28]: Maybe add `title` (we had this for some reason)

**Review by kaitlinbolling — APPROVED** [review]
👍  Looks good! Just left some comments for consideration (I was cross-checking our tags/attrs w/ https://github.com/cure53/DOMPurify/tree/e1c19cf6407d782b666cb1d02a6af191f9cbc09e/src, but I might not be looking in the right place!)

File: packages/core/utils/sanitizeHTML.js (hunk 1 above)
Comment (ostowe) [reply packages/core/utils/sanitizeHTML.js]: Looks like we had `rel`, `href`, and `type` in here for `link` tags, so I'm gonna keep those in for now. afaik ARIA and data attributes are supported out of the box, and configured separately. Will add the other two, though.

**Pull request merged**